src/commands/misc/time.js

```diff
@@ -0,0 +1,54 @@
+const { CommandStructures, SwitchbladeEmbed } = require('../../')
+const { Command, CommandParameters, StringParameter } = CommandStructures
+
+const moment = require('moment-timezone')
+
+// Create a map of tz abbreviations to locations
+// { PDT: [ 'America/Dawson', 'America/Ensenada', 'America/Los_Angeles', ... ], ... }
+const mappedZones = moment.tz.names()
+  .reduce((zones, tzName) => {
+    const abbr = moment.tz(tzName).format('z')
+
+    if (zones[abbr] && zones[abbr] instanceof Array) {
+      zones[abbr].push(tzName)
+    } else {
+      zones[abbr] = [ tzName ]
+    }
+
+    return zones
+  }, {})
+
+module.exports = class Time extends Command {
+  constructor (client) {
+    super(client)
+    this.name = 'time'
+    this.aliases = ['currenttime']
+
+    this.parameters = new CommandParameters(this,
+      new StringParameter({ full: true, missingError: 'commands:time.invalidTimezone' })
+    )
+  }
+
+  run ({ t, author, channel }, requestedTz) {
```

Comment:
excellent work! but try to add the moment locale to it.

```suggestion
  run ({ t, author, channel, guildDocument }, requestedTz) {
  moment.locale(guildDocument.language)
```